ivy/array/general.py

```diff
@@ -1,8 +1,64 @@
 # global
 import abc
-
 # ToDo: implement all methods here as public instance methods
 
+# local
+import ivy
+
 
 class ArrayWithGeneral(abc.ABC):
-    pass
+    def all_equal(
+            self,
+            x,
+            equality_matrix: bool = False
+    ):
+        """Determines whether the inputs are all equal.
```

Comment:
The docstring before the examples need not have the description of the parameters and what's returned, you can refer [this](https://github.com/unifyai/ivy/blob/master/ivy/array/elementwise.py#L34) docstring for the same.